pkg/cmd/populate/populate.go

```diff
@@ -452,7 +452,8 @@ func (o *Options) secretCommandRunner(_ string) (cmdrunner.CommandRunner, cmdrun
 func (o *Options) getSecretManager(backendType string) (secretstore.Interface, error) {
 	store := GetSecretStore(v1alpha1.BackendType(backendType))
 
-	if store == secretstore.SecretStoreTypeVault {
+	isExternalVault := os.Getenv("EXTERNAL_VAULT")
```

Comment:
@ankitm123 mind if you have a followup task to document this new env var once we know the external vault setup is working?

Reply:
Yes, I will document it and also add tests.